### ISSUE TITLE
Null Pointer Exception: Ticking Entity Crash - Zombie

### ISSUE THREAD

**Issue by usafphoenix** [Mithion/ArsMagica2#1442]
http://pastebin.com/ZKtfXA6X

reference to: https://github.com/Mithion/ArsMagica2/issues/1416

as the crash is probably being triggered by the same mechanism.

Used PotionIDConflict resolver mod after fresh pack setup and configuring to resolve potion ID conflicts. Created and Deleted world to ensure proper world gen, did all forge/NEI data dumps, corrected any conflicts i could see, rebooted pack and built a new world. ---crashed on first night when hostiles started spawning.

**Comment by usafphoenix** [comment]
potion dump csv file contains 4000 entries of null (unused potion ID slots).
changing extendedpotions.cfg file to extend the array of potion ID slots to only those used to see if that helps. may do nothing

**Comment by bookerthegeek** [comment]
Crash on world start or client start? 

Logs for new crash
Can you ghist the log of potion ids?
Try with anti Id conflict. Found it works very well.

**Comment by usafphoenix** [comment]
new crash: http://pastebin.com/4yL3bEw6
going to load up anti ID conflict, resolve an errors and come back

**Comment by bookerthegeek** [comment]
Starthe with no configs, and resolve as they come up. Also useful is the forge configs override mod... name escapes me at the moment.

**Comment by usafphoenix** [comment]
resolved the conflicts that popped up but still no fix. what are some modpacks that have AM2, i'd like to compare some mods i'm using against what they have installed and see if removing those solves issue

**Comment by usafphoenix** [comment]
(my idea now is to copy another modpack's configs and only install the mods which are similar between the two packs, then one-by-one add and resolve any conflicts and see if i can spot when the crash begins occuring

**Comment by bookerthegeek** [comment]
Can I get a Dropbox of your mod/config/world save/world folder/etc. In a Dropbox and I'll take a look tonight/tomorrow.

**Comment by BlackSoldierB** [comment]
I am also crashing with the latest version.

http://pastebin.com/mRxGb0z2

I have traced it to the following code in the adventure backpack mod.
Looks like a potion problem.
` for (Iterator iterator = this.mc.thePlayer.getActivePotionEffects()
                        .iterator(); iterator.hasNext(); xPos += BUFF_ICON_SPACING)
                {
                    PotionEffect potioneffect = (PotionEffect) iterator.next();
                    Potion potion = Potion.potionTypes[potioneffect.getPotionID()];

                    if (potion.hasStatusIcon()) <-- Crash here, line 91
                    {
                        int iconIndex = potion.getStatusIconIndex();
                        this.drawTexturedModalRect(
                                xPos, yPos,
                                BUFF_ICON_BASE_U_OFFSET + iconIndex % BUFF_ICONS_PER_ROW * BUFF_ICON_SIZE, BUFF_ICON_BASE_V_OFFSET + iconIndex / BUFF_ICONS_PER_ROW * BUFF_ICON_SIZE,
                                BUFF_ICON_SIZE, BUFF_ICON_SIZE);
                    }
                }`

**Comment by Mithion** [comment]
in the snippet you posted, the variable potion is set on the previous line:
Potion potion = Potion.potionTypes[potioneffect.getPotionID()];

What this means is that  the potions array doesn't have a potion at the index that the potion entry is pointing at.

This is usually caused by potion IDs changing while there is a potion effect active on the player.  If you updated to 1.4.0.009 while your character had a buff/debuff on them, this could happen.

The fix:

Close the game.
Using an NBT editor, remove any potion effects from your character, and in most cases that solves the issue.

**Comment by usafphoenix** [comment]
https://dl.dropboxusercontent.com/u/1867004/CustomPack5.rar

**Comment by BlackSoldierB** [comment]
I am still crashing with the latest version.
This is what I did.
Load version 0008 save.
Removed my armor because it gave me some buffs.
Quit.
Upgrade to 0009.
Load save.
Equip armor.
Crash right after I equipped my armor.

Stacktrace: http://pastebin.com/M1TTMkXF

**Comment by TheIcyOne** [comment]
Do you have ID 11 on again in this save?

**Comment by BlackSoldierB** [comment]
Yeah I did.

**Comment by usafphoenix** [comment]
Going to try and rebuild pack. As was just informed of a mod-interaction with potionIDconflicts and ArsMagica2.. which i had been running into previously.
Per: https://github.com/makeoo/Gadomancy/issues/49#issuecomment-183376512
I'm told the potionIDconflicts mod doesn't like mods that extend on the potions system with HUDs, etc. specifically AM2, StatusEffectHUD---both of which were having problems whenever i was trying to resolve potionID collisions. 
So....manual configuration it is :(

**Comment by ThatOneTraitor** [comment]
@usafphoenix Any Update?

**Comment by usafphoenix** [comment]
no. i don't seem to get this crash unless i forget to remove potionIDconflicts mod before building a world. Which, i, admittingly do quite often as i'm always adding/removing mods and re-enabling it to help resolve conflicts. But since I also often remove several mods (or add several) at a time. for now, i'm content on blaming potionIDconflicts for the mishaps.
